tests/drivers/can/api/src/canfd.c

```diff
@@ -425,6 +544,47 @@ ZTEST_USER(canfd, test_invalid_sample_point)
 	zassert_equal(err, -EINVAL, "invalid sample point of 100.0% accepted (err %d)", err);
 }
 
+/**
+ * @brief Test using an invalid bitrate.
+ */
+ZTEST_USER(canfd, test_invalid_bitrate)
+{
+	struct can_timing timing;
+	int err;
+
+	err = can_calc_timing_data(can_dev, &timing, 0, TEST_SAMPLE_POINT);
+	zassert_equal(err, -EINVAL, "invalid bitrate of 0 bit/s accepted (err %d)", err);
+
+	err = can_calc_timing_data(can_dev, &timing, 8000001, TEST_SAMPLE_POINT);
+	zassert_equal(err, -EINVAL, "invalid bitrate of 8000001 bit/s accepted (err %d)", err);
```

Comment:
I am not sure this warrants an API test.

Reply:
Checks this IF condition
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_common.c#L338